server/src/api/api.ts

```diff
@@ -248,11 +305,131 @@ const resolvers: any = {
+
+        let user;
+        if (users.length === 0) {
+            throw new GraphQLError("Unable to create this request because no user with the UUID provided was found");
+        } else {
+            user = users[0];
+        }
+
+        // fetch the item
+        const item: any = await getItem(args.newRequest.request_item_id, context.user.admin);
+
+        if (!item) {
+            throw new GraphQLError("Can't create request for item that doesn't exist!  Item ID provided: ", args.newRequest.request_item_id);
+        }
+
+        // clip item quantity to allowed values
+        if (args.newRequest.quantity > item.maxRequestQty) {
+            console.log("clipping request quantity (too high), original:", args.newRequest.quantity, ", new:", item.maxRequestQty);
+            args.newRequest.quantity = item.maxRequestQty;
+        } else if (args.newRequest.quantity < 1) {
+            console.log("clipping request quantity (too low), original:", args.newRequest.quantity, ", new:", 1);
+            args.newRequest.quantity = 1;
+        }
+
+        // TODO: if item has approvalRequired == false, initial status should be APPROVED as long as the total qty
+        //       of this item the user has requested
+        let initialStatus: RequestStatus = "SUBMITTED";
+        console.log("item", item);
+        if (!item.approvalRequired) {
+            initialStatus = "APPROVED";
+        }
+
+        // return the request object with the item, censoring price/owner for non-admins
+        let newRequest = await DB.from("requests").insert({
+            ...args.newRequest,
+            status: initialStatus
+        })
+            .returning(["request_id", "created_at", "updated_at"]);
+
+        newRequest = newRequest[0];
+        return {
+            request_id: newRequest.request_id,
+            quantity: args.newRequest.quantity,
+            status: initialStatus,
+            item,
+            user,
+            createdAt: localTimestamp(newRequest.created_at),
+            updatedAt: localTimestamp(newRequest.updated_at)
+        };
+    },
+    deleteRequest: async (root, _args, _context): Promise<boolean> => {
+        // @ts-ignore
+        const {args, context} = fixArguments(root, _args, _context);
+
+        if (!context.user.admin) {
+            throw new GraphQLError("You do not have permission to access the deleteRequest endpoint.");
+        }
+
+        const numRowsAffected = await DB.from("requests")
+            .where({
+                request_id: args.id,
+            })
+            .del();
+
+        return numRowsAffected !== 0;
+    },
+    updateRequest: async (root, _args, _context) => {
+        // @ts-ignore
+        const {args, context} = fixArguments(root, _args, _context);
+
+        if (!context.user.admin) {
+            throw new GraphQLError("You do not have permission to access the updateRequest endpoint.");
```

Comment:
we can punt merging, it can be separate check - update only allowed if admin or request is pending and it's a quantity change